Mathlib/Data/List/InsertIdx.lean

```diff
@@ -19,14 +19,72 @@ open Nat hiding one_pos
 
 namespace List
 
-universe u
+universe u v
 
-variable {α : Type u}
+variable {α : Type u} {β : Type v}
 
 section InsertIdx
 
 variable {a : α}
 
+@[simp]
+theorem sublist_insertIdx (l : List α) (n : ℕ) (a : α) : l <+ (l.insertIdx n a) := by
+  simpa only [eraseIdx_insertIdx] using eraseIdx_sublist (l.insertIdx n a) n
+
+@[simp]
+theorem subset_insertIdx (l : List α) (n : ℕ) (a : α) : l ⊆ l.insertIdx n a :=
+  (sublist_insertIdx ..).subset
+
+/-- Erasing `n`th element of a list, then inserting `a` at the same place
+is the same as setting `n`th element to `a`.
+
+We assume that `n ≠ length l`, because otherwise LHS equals `l ++ [a]` while RHS equals `l`. -/
+@[simp]
```

Comment:
Should this really be simp? It doesn't seem very likely it will find `n ≠ length l`?

Reply:
We have other similar `simp` lemmas like `Function.update_of_ne`.